### PR TITLE
Task 8447033 - automated fix for ADO_Organization_AuthZ_Restrict_Feed_Create_Permission

### PR DESCRIPTION
**Pull request by anandsinghms**
## Description
</br>
 Brief description of the requested change.

## Checklist
- [ ] I have read the instructions mentioned in [Contribute to Code](/Contributing.md).
- [ ] I have read and understood the criteria described under [submitting changes](/Contributing.md#submitting-changes). 
- [ ] The title of the PR clearly describes the intent of the PR.
- [ ] This PR does not introduce any breaking changes to the code.

### CODE REVIEW

File: src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.Organization.ps1 [hunk 1]
```diff
@@ -3684,6 +3684,11 @@ class Organization: ADOSVTBase
                 # Checking if everyone in the organization has create permission on feed
                 $restrictedGroups = @($roleAssignments | Where-Object {"Project Collection Valid Users" -contains $_.split('\')[-1] })
                 $restrictedGroupsCount = $restrictedGroups.Count
+                if ($this.ControlFixBackupRequired)
+                {           
+                    $backup = @{"Descriptor"= $responseObj.identityDescriptor;"Id"=$responseObj.identityId};
```
Comment (juhi037) [inline src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.Organization.ps1:3689]: Automated fix is not working in scenarios where multiple users along with PCVU are given the permissions

File: src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.Organization.ps1 [hunk 2]
```diff
@@ -3703,6 +3708,53 @@ class Organization: ADOSVTBase
...
+                            "role": 1,
+                            "identityDescriptor": "$($($RawDataObjForControlFix.Descriptor).Replace('\','\\'))"                        
+                        }
+"@;                                
+            }
+            else {                                 
+                    if ($body.length -gt 1) {$body += ","}
+                    $body += @"
+                        {                            
+                            "identityId": "$($RawDataObjForControlFix.Id)",
```
Comment (arvindsingh-cloud) [inline src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.Organization.ps1:3735]: Remove duplicate code only Role is changing, other code can be come above if else and role can be assign inside if else.

File: src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.Organization.ps1 [hunk 3]
```diff
@@ -3703,6 +3708,53 @@ class Organization: ADOSVTBase
...
+    hidden [ControlResult] CheckCreatePermissionsForFeedAutomatedFix([ControlResult] $controlResult) {
+
+        try{            
+            $RawDataObjForControlFix = ([ControlHelper]::ControlFixBackup | where-object {$_.ResourceId -eq $this.ResourceId}).DataObject   
+            # roleid : 1 (remove broader group permisson to Create Feed) , roleid : 2 (add broader group permisson to Create Feed)                    
+            $body = "["
+
+            if (-not $this.UndoFix)
+            {                                 
+                    if ($body.length -gt 1) {$body += ","}
```
Comment (arvindsingh-cloud) [inline src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.Organization.ps1:3722]: ??  if ($body.length -gt 1) this condition will never be true

File: src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.Organization.ps1 [hunk 4]
```diff
@@ -3680,10 +3680,17 @@ class Organization: ADOSVTBase
             
                 $groups = $response.Content | Convertfrom-json
                 $roleAssignments = $groups.results.identities.displayname
-
+                foreach ($role in $roleAssignments) {
+                    
+                }
```
Comment (arvindsingh-cloud) [inline src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.Organization.ps1:3685]: Remove empty loop 
foreach ($role in $roleAssignments) {

                }

File: src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.Organization.ps1 [hunk 5]
```diff
@@ -3703,6 +3710,44 @@ class Organization: ADOSVTBase
...
+            # roleid : 1 (remove broader group permisson to Create Feed) , roleid : 2 (add broader group permisson to Create Feed)                    
+            $body = "["
+            $roleId = @()
+                if (-not $this.UndoFix){
+                    $roleId =1;
+                }
+                else{
+                    $roleId =2;                    
+                }
+                if ($body.length -gt 1) {$body += ","}
```
Comment (arvindsingh-cloud) [inline src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.Organization.ps1:3728]: if ($body.length -gt 1) {$body += ","} this condition will never come true,

**Review by v-soukum — APPROVED** [review]

**Review by arvindsingh-cloud — APPROVED** [review]

**Pull request merged**